### PR TITLE
DetailedErrorsToClient - only send detailed error info if this is on

### PR DESCRIPTION
**Pull request by GuyLewin**

### CODE REVIEW

File: server/service.go [hunk 1]
```diff
@@ -331,9 +335,13 @@ func (svr *Service) HandleListener(l net.Listener) {
 				case *msg.NewVisitorConn:
 					if err = svr.RegisterVisitorConn(conn, m); err != nil {
 						xl.Warn("register visitor conn error: %v", err)
+						errStr := "register visitor conn error"
```
Comment (fatedier) [inline server/service.go]: Wrap a new function for this?

```golang
func GenerateResponseErrorString(summary string, err error, detailed bool) string {
    if detailed {
        ...
    } else {
        ...
    }
}
```

**Comment by fatedier** [comment]
It has conflicts with #1643.

File: server/service.go (hunk 1 above)
Comment (GuyLewin) [reply server/service.go]: Good point, will do

**Comment by GuyLewin** [comment]
@fatedier  - what do you think, should we pass `xl` into the function and it'll log the warning too? Or is it too much coupling and we should keep it as is?

**Comment by fatedier** [comment]
@GuyLewin 
> what do you think, should we pass xl into the function and it'll log the warning too? Or is it too much coupling and we should keep it as is?

I think it's ok now.

**Pull request merged**